server/src/test/java/org/opensearch/tasks/consumer/TopNSearchTasksLoggerTests.java

```diff
@@ -59,22 +80,65 @@ public void testLoggerWithTasks() {
 
     public void testLoggerWithHighFrequency() {
         // setting the frequency to a really large value and confirming that nothing gets written to log file.
-        final Settings settings = Settings.builder().put(LOG_TOP_QUERIES_SIZE, 1).put(LOG_TOP_QUERIES_FREQUENCY, "10m").build();
-        topNSearchTasksLogger = new TopNSearchTasksLogger(settings);
+        final Settings settings = Settings.builder()
+            .put(LOG_TOP_QUERIES_SIZE_SETTING.getKey(), 1)
+            .put(LOG_TOP_QUERIES_FREQUENCY_SETTING.getKey(), "10m")
+            .build();
+        topNSearchTasksLogger = new TopNSearchTasksLogger(settings, null);
         generateTasks(5);
         generateTasks(2);
 
         assertNull(appender.getLastEventAndReset());
     }
 
+    public void testDynamicSettings() {
+        Settings settings = Settings.builder()
+            .put(LOG_TOP_QUERIES_SIZE_SETTING.getKey(), 1)
+            .put(LOG_TOP_QUERIES_FREQUENCY_SETTING.getKey(), "60s")
+            .build();
+        ClusterSettings clusterSettings = new ClusterSettings(Settings.EMPTY, ClusterSettings.BUILT_IN_CLUSTER_SETTINGS);
+        topNSearchTasksLogger = new TopNSearchTasksLogger(settings, clusterSettings);
+        settings = Settings.builder().put(LOG_TOP_QUERIES_SIZE_SETTING.getKey(), "10").build();
+        ClusterUpdateSettingsResponse response = client().admin().cluster().prepareUpdateSettings().setPersistentSettings(settings).get();
+        assertEquals(response.getPersistentSettings().get(LOG_TOP_QUERIES_SIZE_SETTING.getKey()), "10");
+
+        settings = Settings.builder().put(LOG_TOP_QUERIES_FREQUENCY_SETTING.getKey(), "100s").build();
+        response = client().admin().cluster().prepareUpdateSettings().setPersistentSettings(settings).get();
+        assertEquals(response.getPersistentSettings().get(LOG_TOP_QUERIES_FREQUENCY_SETTING.getKey()), "100s");
+    }
+
+    public void testMinMaxSettingValues() {
+        final Settings settings = Settings.builder()
+            .put(LOG_TOP_QUERIES_SIZE_SETTING.getKey(), 1)
+            .put(LOG_TOP_QUERIES_FREQUENCY_SETTING.getKey(), "60s")
+            .build();
+        topNSearchTasksLogger = new TopNSearchTasksLogger(settings, null);
+        final Settings sizeSetting = Settings.builder().put(LOG_TOP_QUERIES_SIZE_SETTING.getKey(), "50").build();
+        // request should fail as it crosses maximum defined value(25)
+        expectThrows(
+            IllegalArgumentException.class,
+            () -> client().admin().cluster().prepareUpdateSettings().setPersistentSettings(sizeSetting).get()
+        );
+
+        final Settings freqSetting = Settings.builder().put(LOG_TOP_QUERIES_FREQUENCY_SETTING.getKey(), "30s").build();
+        // request should fail as it crosses minimum defined value(60s)
+        expectThrows(
+            IllegalArgumentException.class,
+            () -> client().admin().cluster().prepareUpdateSettings().setPersistentSettings(freqSetting).get()
+        );
+    }
```

Comment:
I feel that these tests doesn't belong here. They're not testing anything about the feature in this PR, instead, they're testing the functionality of the common `Settings`' framework.